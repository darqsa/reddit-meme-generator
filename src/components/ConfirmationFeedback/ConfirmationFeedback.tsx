import {
  ConfirmationButtonContainerStyled,
  ConfirmationButtonStyled,
  ConfirmationContainerStyled,
  ConfirmationTextStyled,
} from "./ConfirmationFeedbackStyled";

const confirmationFeedback = (): JSX.Element => {
  return (
    <ConfirmationContainerStyled>
      <ConfirmationTextStyled>
        Are you sure you want to delete this meme?
      </ConfirmationTextStyled>
      <ConfirmationButtonContainerStyled>
        <ConfirmationButtonStyled>Yes</ConfirmationButtonStyled>
        <ConfirmationButtonStyled>No</ConfirmationButtonStyled>
      </ConfirmationButtonContainerStyled>
    </ConfirmationContainerStyled>
  );
};
export default confirmationFeedback;
