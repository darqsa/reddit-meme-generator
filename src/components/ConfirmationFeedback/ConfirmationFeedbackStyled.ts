import styled from "styled-components";

const ConfirmationContainerStyled = styled.div`
  @media screen and (min-width: 700px) {
    width: 500px;
  }
  align-items: center;
  display: flex;
  flex-direction: column;
  border-radius: 20px;
  background-color: white;
  border: 3px solid red;
  height: 300px;
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  margin: auto;
  justify-content: center;
`;
const ConfirmationTextStyled = styled.span`
  color: red;
  text-align: center;
  font-size: 25px;
  margin: 20px 0;
  font-weight: normal;
`;
const ConfirmationButtonContainerStyled = styled.div`
  display: flex;
  gap: 10px;
`;
const ConfirmationButtonStyled = styled.button`
  border: 2px solid red;
  background: 0%;
  color: red;
  border-radius: 10px;
  width: 120px;
  height: 50px;
  font-family: "Inter", sans-serif;
  font-size: 20px;
  transition: 0.3s;
  &:hover {
    color: white;
    cursor: pointer;
    background-color: red;
  }
`;

export {
  ConfirmationButtonContainerStyled,
  ConfirmationContainerStyled,
  ConfirmationTextStyled,
  ConfirmationButtonStyled,
};
