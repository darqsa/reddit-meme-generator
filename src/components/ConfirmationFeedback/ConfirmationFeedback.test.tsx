import TestRenderer from "react-test-renderer";
import ConfirmationFeedback from "./ConfirmationFeedback";

describe("Given a ConfirmationFeedback component", () => {
  describe("When it's rendered", () => {
    test("Then it should show the ConfirmationFeedback component", () => {
      const expectedConfirmation = TestRenderer.create(
        <ConfirmationFeedback />
      );
      expect(expectedConfirmation).toMatchSnapshot();
    });
  });
});
