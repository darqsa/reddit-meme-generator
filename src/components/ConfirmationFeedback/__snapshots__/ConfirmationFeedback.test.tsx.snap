// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a ConfirmationFeedback component When it's rendered Then it should show the ConfirmationFeedback component 1`] = `
<div
  className="sc-bczRLJ bXeUiX"
>
  <span
    className="sc-gsnTZi kACbis"
  >
    Are you sure you want to delete this meme?
  </span>
  <div
    className="sc-dkzDqf csSjQT"
  >
    <button
      className="sc-hKMtZM fUSGxr"
    >
      Yes
    </button>
    <button
      className="sc-hKMtZM fUSGxr"
    >
      No
    </button>
  </div>
</div>
`;
